### PR TITLE
Microservice for applying migrations automatically

### LINKED ISSUE #1295: Microservice for applying migrations automatically.
**Is your feature request related to a problem? Please describe.**
For our upcoming 0.10.0 update i noticed, that it is very difficult to solve everything in pure SQL.

**Describe the solution you'd like**
I propose adding a new table to the database, which tracks the current migrated version.
Also adding a microservice, which compares the migrated version and the installed version and automatically applies all migrations from the last migrated to the installed version.
Afterwards it sets the last migrated version to the installed version.

### PR DESCRIPTION
**Pull request by Scarjit**
# Description

This pr includes a microservice to apply migrations automatically depending on last and current version

Fixes #1295

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [x] migration_test.go runs, when supplying the necessary migrations and a db
- [x] All sql scripts where tested against a postgresql db

# Checklist:

- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: deployment/united-manufacturing-hub/values.yaml [hunk 1]
```diff
@@ -338,6 +338,20 @@ _001_customMicroservices:
...
+### automigrate ###
+automigrate:
+  enabled: true
+  image:
+    repository: unitedmanufacturinghub/automigrate
+    pullPolicy: IfNotPresent
+  resources:
+    requests:
+      memory: 2Gi
+      cpu: 1000m
```
Comment (JeremyTheocharis) [inline deployment/united-manufacturing-hub/values.yaml:350]: This is very high and might result in the system being unable to start on amaller development setups

File: golang/cmd/automigrate/migrations/migration.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	lastVersion := oldVersion
+	for l != nil {
+		zap.S().Infof("Applying migration %s", l.version)
+		err := l.migration(db)
+		if err != nil {
+			return err
+		}
+		// Set migration as applied
+		_, err = db.Exec(
+			"INSERT INTO migrationtable (frommajor, fromminor, frompatch, tomajor, tominor, topatch, timestamp) VALUES (?, ?, ?, ?, ?, ?, now())",
```
Comment (JeremyTheocharis) [inline golang/cmd/automigrate/migrations/migration.go:219]: Document in documentation :)

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

File: golang/cmd/automigrate/migrations/migration.go (hunk 2 above)
Comment (Scarjit) [reply golang/cmd/automigrate/migrations/migration.go:219]: See:
https://www.notion.so/united-manufacturing-hub/ab58fbca87834bfeb7f654af1dbb43fd?v=61b50b6953fd415bb2ac4dc5dccfc109&p=2d72668a371248ff801869f9fecf648f&pm=s

Comment (Scarjit) [reply golang/cmd/automigrate/migrations/migration.go:219]: Documented in upgrade guide

File: deployment/united-manufacturing-hub/values.yaml (hunk 1 above)
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml:353]: Reduced

**Review by JeremyTheocharis — APPROVED** [review]

**Pull request merged**